curator-framework/src/main/java/org/apache/curator/framework/imps/ProtectedMode.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.curator.framework.imps;
+
+import org.apache.zookeeper.CreateMode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.UUID;
+
+/**
+ * manage the protected mode state for {@link org.apache.curator.framework.imps.CreateBuilderImpl}
+ */
+class ProtectedMode
+{
+    private final Logger log = LoggerFactory.getLogger(getClass());
+    private boolean doProtected = false;
+    private String protectedId = null;
+    private long sessionId = 0;
+
```

Comment:
ProtectedMode is accessed by potentially more than one thread, no? In that case, doesn't it need to use some concurrency synchronization mechanism like a lock or volatile for it's fields, at least to assure visibility?